cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/GroupByAggregationDefinition.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * Specifies how a group by operation should be executed.
+ */
+public class GroupByAggregationDefinition extends AggregationDefinition {
+  private GroupByAggregationDefinition(List<Expression> groupByExpressions, List<Expression> selectExpressions) {
+    super(groupByExpressions, selectExpressions);
+  }
+
+  /**
+   * @return A builder to create a GroupByAggregationDefinition.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Builds a GroupByAggregationDefinition using fields to group by and fields to select.
+   * The fields to select must be specified, while the fields to group by are optional.
+   */
+  public static class Builder {
+    private List<Expression> groupByExpressions;
+    private List<Expression> selectExpressions;
+
+    public Builder() {
+      groupByExpressions = Collections.emptyList();
+      selectExpressions = Collections.emptyList();
+    }
+
+    public Builder groupBy(List<Expression> groupByExpressions) {
```

Comment:
Please also add javadocs to the builder. It helps when working from IDE and building the definition

Reply:
Added.